lib/failure.sh

```diff
@@ -27,6 +27,20 @@ fail_invalid_package_json() {
   fi
 }
 
+fail_dot_heroku_node() {
+  if [ -f "${1:-}/.heroku/node" ]; then
+    header "Build failed"
+    warn "The directory .heroku/node could not be created
```

Comment:
There isn't any public documentation that I can find that describes the not stepping on the `.heroku` directory. Perhaps I should write something short to link to?